### PR TITLE
docs(stat): add stat theming docs

### PR DESCRIPTION
**Pull request by BigDog1400**
Closes https://github.com/chakra-ui/chakra-ui-docs/issues/1089

📝 Description
Adding theming docs for the Stat component

### CODE REVIEW

**Comment by nikolovlazar** [comment]
@BigDog1400 thanks for opening this PR! I can see you haven't provided a showcase example, and a custom variant. Do you plan on providing those examples?

File: content/docs/components/stat/theming.mdx [hunk 1]
```diff
@@ -3,4 +3,104 @@ id: stat
...
 ---
 
-TODO
+The Stat component is a multipart component. The styling needs to be applied to each part specifically.
+
+> To learn more about styling multipart components, visit the
+> [Component Style](/docs/styled-system/component-style#styling-multipart-components)
+> page.
+
+This component has five parts: `container`, `label`, `helpText`, `icon`,  and `number`. You can find more information in the source for the styles [here](https://github.com/chakra-ui/chakra-ui/blob/main/packages/components/theme/src/components/stat.ts).
```
Comment (nikolovlazar) [inline content/docs/components/stat/theming.mdx:12]: I've prepared an anatomy illustration for you. You can grab the code [from this gist](https://gist.github.com/nikolovlazar/ee0d3372065ca045a24bb8d1694e0520).

Also, make a new section for the anatomy:
```markdown
## Anatomy

- A: `container`
- B: `label`
- C: `helpText`
- D: `icon`
- E: `number`

[PLACE THE ILLUSTRATION CODE HERE]
```

**Review by nikolovlazar — CHANGES_REQUESTED** [review]

**Comment by BigDog1400** [comment]
> @BigDog1400 thanks for opening this PR! I can see you haven't provided a showcase example, and a custom variant. Do you plan on providing those examples?

Sure! I will try to update it today or tomorrow.

File: content/docs/components/stat/theming.mdx (hunk 1 above)
Comment (BigDog1400) [reply content/docs/components/stat/theming.mdx:12]: Thanks, i will include it

**Comment by BigDog1400** [comment]
Hi @nikolovlazar ! The requested changes have already been made. Let me know if something is missing or wrong

**Review by nikolovlazar — APPROVED** [review]
There are some minor sandbox content changes to be done, but I'll take care of that. Everything else looks good. Thanks for contributing to the Stat theming docs @BigDog1400 🙌

**Comment by nikolovlazar** [comment]
@allcontributors please add @BigDog1400 for docs.

**Pull request merged**